### PR TITLE
chore: update marked dependencies

### PR DESCRIPTION
**Pull request by targos**
The TypeScript compilation looks good (everything is still CommonJS and the dynamic imports are not transformed)
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,7 +58,8 @@
     "mrm": "^4.0.0",
     "np": "^7.6.1",
     "prettier": "^2.6.1",
-    "typescript": "^4.6.3",
+    "ts-node": "^10.7.0",
+    "typescript": "^4.7.0-dev.20220329",
```
Comment (targos) [inline package.json]: "Node12" is only enabled in nightly TS builds for now.

**Comment by targos** [comment]
```
> node -r ts-node/register/transpile-only bin/test.ts

(node:1655) UnhandledPromiseRejectionWarning: Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/runner/work/sink/sink/src/disableLogger.js' imported from /home/runner/work/sink/sink/bin/test.ts
```

I don't have this error locally with Node 16 and 17.

**Comment by targos** [comment]
```
> @adonisjs/sink@5.2.2 test
> node -r ts-node/register/transpile-only bin/test.ts


   Unknown file extension ".ts" for /Users/targos/git/adonisjs/adonis-sink/test/adonis-rc-file.spec.ts
```

I have this error locally, though. It seems to come from Node's ESM resolver, which doesn't make sense to me because we do not have `"type": "module"` in package.json, so the CommonJS resolver should be used here 🤔

**Comment by targos** [comment]
https://github.com/adonisjs/sink/pull/34/commits/f777eb42fd1294e9165c0fef9aa05b17adf42ab1 fixes the above errors but introduces other errors.
It's probably related to ts-node or/and japa. I don't know those tools enough to say.

**Comment by thetutlage** [comment]
The main issue here is TypeScript compiling the dynamic imports to `require` statements :/

**Comment by targos** [comment]
No, the issue is that `ts-node` doesn't like imports with the extension, because they point to files that do not exist.

**Comment by thetutlage** [comment]
The `ts-node` issue happens when you are running the tests. But I am talking about when you compile the TypeScript is also rewriting the imports to require. So there were 2 issues

**Pull request closed without merging**